packages/components/src/internal/components/samples/SampleAliquotsGridPanel.tsx

```diff
@@ -0,0 +1,157 @@
+        const { sampleLsid, schemaQuery, actions, user, rootLsid, inventoryCols } = this.props;
+
+        const queryConfig = getSampleAliquotsQueryConfig(
+            schemaQuery.getQuery(),
+            sampleLsid,
+            true,
+            rootLsid,
+            getOmittedSampleTypeColumns(user, inventoryCols)
+        );
+        // don't need to load the data here because that is done by default in the GridPanel.
+        actions.addModel(queryConfig, false);
+    }
+
+    getQueryModel(): QueryModel {
+        return Object.values(this.props.queryModels)[0];
+    }
+
+    afterAction = () => {
+        const { actions, schemaQuery, onSampleChangeInvalidate } = this.props;
+        this.resetState();
+        onSampleChangeInvalidate(schemaQuery);
+        const model = this.getQueryModel();
+        actions.loadModel(model.id, true);
+    };
+
+    hideConfirm = () => {
+        this.setState(() => ({ showConfirmDelete: false }));
+    };
+
+    onDelete = () => {
+        if (this.hasSelection()) {
+            this.setState(() => ({ showConfirmDelete: true }));
+        }
+    };
+
+    resetState = () => {
+        if (this.hasSelection()) {
+            this.setState(() => ({ showConfirmDelete: false }));
+        }
+    };
+
+    hasSelection() {
+        return this.getQueryModel().selections?.size > 0;
+    }
+
+    renderButtons = () => {
+        const queryModel = this.getQueryModel();
+        const Node = this.props.storageButton;
+        return (
+            <div className="btn-group">
+                <RequiresPermission perms={PermissionTypes.Delete}>
+                    <ManageDropdownButton id="samplealiquotlisting">
+                        <SelectionMenuItem
+                            id="sample-aliquot-delete-menu-item"
+                            text="Delete Aliquots"
+                            onClick={() => this.onDelete()}
+                            queryModel={queryModel}
+                            nounPlural="aliquots"
+                        />
+                    </ManageDropdownButton>
+
+                    {Node && (
+                        <Node user={this.props.user} queryModel={queryModel} afterStorageUpdate={this.afterAction} />
+                    )}
+                </RequiresPermission>
+            </div>
+        );
+    };
+
+    renderDeleteModal() {
+        const model = this.getQueryModel();
+
+        if (!this.state.showConfirmDelete) return null;
+
+        return (
+            <EntityDeleteModal
+                queryModel={model}
+                useSelected={true}
+                afterDelete={this.afterAction}
+                onCancel={this.resetState}
+                entityDataType={SampleTypeDataType}
+                verb="deleted and removed from storage"
+            />
+        );
+    }
+
+    render() {
+        const { actions } = this.props;
+
+        const model = this.getQueryModel();
+
+        if (!model) return <LoadingSpinner />;
```

Comment:
One advantage of using `withQueryModels` and passing the query configuration into the component (rather than calling `addModels`) is that you're guaranteed the query model instance will exist. This reduces the need for being defensive against it not existing. Here is what that might look like:

```tsx
interface Props {
    onSampleChangeInvalidate: (schemaQuery: SchemaQuery) => void;
    storageButton?: StorageButton;
    user: User;
}

interface State {
    showConfirmDelete: boolean;
}

export class SampleAliquotsGridPanelImpl extends PureComponent<Props & InjectedQueryModels, State> {

	// Implementation details...

    render() {
        const { actions, queryModels, storageButton, user } = this.props;

        // You can count on "model" to exist on "queryModels"
        const { model } = queryModels;

        return (
            <>
                <GridPanel
                    actions={actions}
                    ButtonsComponent={AliquotGridButtons}
                    buttonsComponentProps={{
                        afterAction: this.afterAction,
                        onDelete: this.onDelete,
                        StorageButtonsComponent: storageButton,
                        user,
                    }}
                    loadOnMount={false}
                    model={model}
                    showViewMenu={false}
                />

                {this.state.showConfirmDelete && (
                    <EntityDeleteModal
                        afterDelete={this.afterAction}
                        entityDataType={SampleTypeDataType}
                        onCancel={this.resetState}
                        queryModel={model}
                        useSelected
                        verb="deleted and removed from storage"
                    />
                )}
            </>
        );
    }
}

const SampleAliquotsGridPanelWithModel = withQueryModels<Props>(SampleAliquotsGridPanelImpl);

interface SampleAliquotsGridPanelProps extends Props {
    inventoryCols?: string[];
    rootLsid?: string; // if sample is an aliquot, use the aliquot's root to find subaliquots
    sampleLsid: string;
    schemaQuery: SchemaQuery;
}

export const SampleAliquotsGridPanel: FC<SampleAliquotsGridPanelProps> = props => {
    const { inventoryCols, sampleLsid, schemaQuery, rootLsid, user } = props;

    const queryConfigs = {
        model: getSampleAliquotsQueryConfig(
            schemaQuery.getQuery(),
            sampleLsid,
            true,
            rootLsid,
            getOmittedSampleTypeColumns(user, inventoryCols)
        ),
    };

    return <SampleAliquotsGridPanelWithModel {...props} autoLoad queryConfigs={queryConfigs} />;
};

SampleAliquotsGridPanel.displayName = 'SampleAliquotsGridPanel';
```

Reply:
Done (in patch)